tests/typescript/conformance/classes/classDeclarations/classAbstractKeyword/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -0,0 +1,889 @@
+var c: C;
+
+a = new B();
+b = new B();
+c = new B();
+
+`;
+
+exports[`classAbstractInstantiations2.ts 1`] = `
+class A {
+}
+
+abstract class B {
+    foo(): number { return this.bar(); }
+    abstract bar() : number;
+}
+
+new B;
+
+var BB: typeof B = B;
+var AA: typeof A = BB;
+new AA;
+
+function constructB(Factory : typeof B) {
+    new Factory;
+}
+
+var BB = B;
+new BB;
+
+var x : any = C;
+new x;
+
+class C extends B { }
+
+abstract class D extends B { }
+
+class E extends B {
+    bar() { return 1; }
+}
+
+abstract class F extends B {
+    abstract foo() : number;
+    bar() { return 2; }
+}
+
+abstract class G {
+    abstract qux(x : number) : string;
+    abstract qux() : number;
+    y : number;
+    abstract quz(x : number, y : string) : boolean;
+
+    abstract nom(): boolean;
+    nom(x : number): boolean;
+}
+
+class H {
+    abstract baz() : number;
+}~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+class A {}
+
+abstract class B {
+  foo(): number {
+    return this.bar();
+  }
+  abstract bar(): number;
+}
+
+new B();
+
+var BB: typeof B = B;
+var AA: typeof A = BB;
+new AA();
+
+function constructB(Factory: typeof B) {
+  new Factory();
+}
+
+var BB = B;
+new BB();
+
+var x: any = C;
+new x();
+
+class C extends B {}
+
+abstract class D extends B {}
+
+class E extends B {
+  bar() {
+    return 1;
+  }
+}
+
+abstract class F extends B {
+  abstract foo(): number;
+  bar() {
+    return 2;
+  }
+}
+
+abstract class G {
+  abstract qux(x: number): string;
+  abstract qux(): number;
+  y: number;
+  abstract quz(x: number, y: string): boolean;
+
+  abstract nom(): boolean;
+  nom(x: number): boolean;
+}
+
+class H {
+  abstract baz(): number;
+}
+
+`;
+
+exports[`classAbstractMethodInNonAbstractClass.ts 1`] = `
+class A {
+    abstract foo();
+}
+
+class B {
+    abstract foo() {}
+}~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+class A {
+  abstract foo();
+}
+
+class B {
+  abstract foo() {}
+}
+
+`;
+
+exports[`classAbstractMethodWithImplementation.ts 1`] = `
+abstract class A {
+    abstract foo() {}
+}~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+abstract class A {
+  abstract foo() {}
+}
+
+`;
+
+exports[`classAbstractMixedWithModifiers.ts 1`] = `
+abstract class A {
+    abstract foo_a();
+    
+    public abstract foo_b();
+    protected abstract foo_c();
+    private abstract foo_d();
+    
+    abstract public foo_bb();
+    abstract protected foo_cc();
+    abstract private foo_dd();
+    
+    abstract static foo_d();
+    
+    static abstract foo_e();
+}~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+abstract class A {
+  abstract foo_a();
+
+  public abstract foo_b();
+  protected abstract foo_c();
+  private abstract foo_d();
+
+  public abstract foo_bb();
+  protected abstract foo_cc();
+  private abstract foo_dd();
+
+  static abstract foo_d();
+
+  static abstract foo_e();
```

Comment:
Yeah it's neat:

![pretty-typescript](https://cloud.githubusercontent.com/assets/1297597/25421907/43c4d62a-2aa2-11e7-88d9-f3df794f9bf8.gif)